### PR TITLE
Schema queries and mutations added

### PR DESCRIPTION
**Pull request by aye-dev**
Schema queries and mutations added

### CODE REVIEW

File: backend/schema.js [hunk 1]
```diff
@@ -14,71 +15,151 @@ const typeDefs = gql`
...
-    allUsers: [User]
+  #QUERIES#####################################
+
+  #based on token
+  query {
+    me: User
+  }
+
+  query Team($id: Int) {
+    team(id: $id): Team
```
Comment (ValterSantosMatos) [inline backend/schema.js]: You can write all queries in a single object like:
See https://github.com/apollographql/apollo-server-tutorial/blob/master/data/schema.js

```
query {
    me: User
    team(id: $id): Team
    ...
  }
```

File: backend/schema.js [hunk 2]
```diff
@@ -14,71 +15,151 @@ const typeDefs = gql`
...
+  }
+
+  query Job($id: Int) {
+    job(id: $id): Job
+  }
+
+  query MatchingTags ($search: String, $first: Int, $cursor: String) {
+    matches(search: $search, first: $first, cursor: $cursor): {
+      totalCount
```
Comment (ValterSantosMatos) [inline backend/schema.js]: This could be its own Type (it will be easier to understand and introspect the code)

File: backend/schema.js [hunk 3]
```diff
@@ -14,71 +15,151 @@ const typeDefs = gql`
...
+    header
+    description
+    tags
+  }
+
+  #MUTATIONS###################################
+  
+  # the only upsertable user is the authenticated one 
+  mutation UpsertUser ($user: User){
+    upsert(user: $user): User
```
Comment (ValterSantosMatos) [inline backend/schema.js]: You can add all in the same object (same as queries)
Input types are different in GraphQl, you can't use `(user: $user)`, see https://graphql.org/learn/schema/#input-types

File: backend/schema.js [hunk 4]
```diff
@@ -14,71 +15,151 @@ const typeDefs = gql`
...
+
+  fragment TeamSummary on Team {
+    id
+    name  
+    header
+    description
+    tags
+  }
+
+  #MUTATIONS###################################
```
Comment (ValterSantosMatos) [inline backend/schema.js]: Naming wise I tend to follow [this convention](https://blog.apollographql.com/designing-graphql-mutations-e09de826ed97), although its not needed at all for this project, but so you know :) In the end everything looks like:
```
type RootMutation {
  createTodo(input: CreateTodoInput!): CreateTodoPayload
  toggleTodoCompleted(input: ToggleTodoCompletedInput!): ToggleTodoCompletedPayload
  updateTodoText(input: UpdateTodoTextInput!): UpdateTodoTextPayload
  completeAllTodos(input: CompleteAllTodosInput!): CompleteAllTodosPayload
}
```

**Comment by ValterSantosMatos** [comment]
Btw, es um fofo @aye-dev

File: data/readme.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#failing apollo client:codegen src/types --target=flow
```
Comment (Mvii7) [inline data/readme.txt]: Change to MD pls

**Pull request merged**